### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by jschofield1**

### CODE REVIEW

**Comment by kaykay38** [comment]
I will review this.

**Comment by kaykay38** [comment]
# Reading Assignment
Chapters 10 and 11

# Assignment

- `Jester` class takes in the correct interfaces: ✔
- `Jester` dependencies are null checked: ✔
- `Jester` appropriately retrieves jokes and sends them to the output: ✔
- Chuck Norris jokes gets filtered out: ✔
- Jester code coverage is above 90%:  ✔ 93.27%

### Extra Credit
- Output to console implementation is unit tested: ❌
- `JokeService` uses JSON: ✔

Good job!

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+
+namespace CanHazFunny
+{
+    public class Jester
+    {
+        private IJokeService jokeService;
+        private IJokeOutput jokeOutput;
```
Comment (Keboo) [inline CanHazFunny/CanHazFunny/Jester.cs:8]: Field should be underscore PascalCase
```suggestion
        private IJokeService _JokeService;
        private IJokeOutput _JokeOutput;
```

File: CanHazFunny/CanHazFunny/JokeService.cs [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
 {
-    public class JokeService
+    public class JokeService : IJokeService
     {
         private HttpClient HttpClient { get; } = new();
 
         public string GetJoke()
         {
-            string joke = HttpClient.GetStringAsync("https://geek-jokes.sameerkumar.website/api").Result;
+            string joke = HttpClient.GetStringAsync("https://geek-jokes.sameerkumar.website/api?format=json").Result;
```
Comment (Keboo) [inline CanHazFunny/CanHazFunny/JokeService.cs:11]: This is good, but doesn't actually parse the Json string.

**Review by Keboo — COMMENTED** [review]
Very clean code. Nice work

**Pull request closed without merging**